### ISSUE TITLE
Anyone having problems with Blue Coffee and other themes?

### ISSUE THREAD

**Issue by Qwart376** [leftwm/leftwm-community-themes#21]
I've recently installed Arch on my new laptop. When I tried to install my theme through leftwm-theme, it turned out to be broken until I swapped files with the coffee.

Theme works normally when downloaded directly from github, or when I swap files with other working themes (also apply to other themes made by community, not the stock ones)

I'll be switching from Arco to mainline Arch and try that again. Just want to ask if other people have the same problem

**Comment by Qwart376** [comment]
Since I don't want to create another topic: What is a keybind for pulse audio (volume up/down/mute)?

**Comment by mautamu** [comment]
To answer the second question first:

> Since I don't want to create another topic for something this small: What is a keybind for pulse audio (volume up/down/mute)?

https://github.com/leftwm/leftwm/blob/003c9cba8f27b61ec2dba6b051c32a4b583baed1/src/utils/xkeysym_lookup.rs#L1047

If you want the volume up/down on the keyboard to work, it’s the key linked above. Just make sure to give it an empty modifier in your configuration and a value of (pulse audio command).

**Comment by mautamu** [comment]
As for the theme itself: 

Is there something missing when installing from leftwm-theme?

**Comment by Qwart376** [comment]
> As for the theme itself:
> 
> Is there something missing when installing from leftwm-theme?

No, everything is place. I switched the files with Coffee and everything worked fine. I'll do tests tomorrow and see what happens

**Comment by Qwart376** [comment]
Same issue on fresh Arch. Can you install Blue Coffee and see if it works on your side?

**Comment by mautamu** [comment]
Only able to reproduce with Blue Coffee theme. Error is on line 4 of up script. Additional errors on down script.

**Comment by taylor85345** [comment]
leftwm-theme clones your theme from the Blue Coffee submodule in the leftwm-community-themes repository, which is a branch of your repo reflecting it's state at the time you submitted the original pull request. Any changes made after your initial submission will not be reflected by a leftwm-theme install until you update the Blue Coffee submodule in leftwm-community-themes, which may be why the theme is installing in a broken state

**Comment by taylor85345** [comment]
Also, it looks like your `up` script isn't downloading with execute permissions. `chmod +x up` fixed the issue for me and allowed the theme to load normally

**Comment by Qwart376** [comment]
Sorry for late reply, but my HDD died

> Also, it looks like your `up` script isn't downloading with execute permissions. `chmod +x up` fixed the issue for me and allowed the theme to load normally

Thanks for the answer, I was told that leftwm-theme automatically sync with main branch. Can you tell me more about the` up` script issue and how to update submodule?

**Comment by taylor85345** [comment]
You might actually be right. The submodule in the repository is always a little behind, but  `leftwm-theme install` actually seems to pull from main, though `upgrade` doesn't seem to do anything.

To start with, you might just try adding execute permissions to the `up` script and pushing the change back to your github.

If you do need to update the submodule, which I'm now less confident about, you'll need to fork the leftwm-community-themes and clone it like you did when you originally submitted (see the wiki).Once you've cloned the repo, do the following:

`git submodule update --init 'Blue Coffee'`
`cd 'Blue Coffee`
`git checkout main`
`git pull`
`git commit -a`

You may also want to update the version number in `known.toml` for your theme, but not sure if that's necessary. From there just `git push` the changes back to github and submit a pull request (again, per the wiki).

**Comment by mautamu** [comment]
> though upgrade doesn't seem to do anything.
Ah, I will file an issue over on leftwm/leftwm-theme about this. It is UB to do that one would suppose. Not sure how best to handle it.

**Comment by Qwart376** [comment]
> You might actually be right. The submodule in the repository is always a little behind, but `leftwm-theme install` actually seems to pull from main, though `upgrade` doesn't seem to do anything.
> 
> To start with, you might just try adding execute permissions to the `up` script and pushing the change back to your github.
> 
> If you do need to update the submodule, which I'm now less confident about, you'll need to fork the leftwm-community-themes and clone it like you did when you originally submitted (see the wiki).Once you've cloned the repo, do the following:
> 
> `git submodule update --init 'Blue Coffee'`
> `cd 'Blue Coffee`
> `git checkout main`
> `git pull`
> `git commit -a`
> 
> You may also want to update the version number in `known.toml` for your theme, but not sure if that's necessary. From there just `git push` the changes back to github and submit a pull request (again, per the wiki).

Do you know what exactly to add to the `up` script? I simply forked the Coffee and the only thing changed in up script was using feh for background (which I borrowed from Dracula Rounded)

**Comment by taylor85345** [comment]
Don't add anything to it. Just run `chmod +x up` from the terminal and that will add the permissions. If you're uploading files to github through a browser, that probably doesn't preserve permissions, so you'll want to `git push` the changes back from the terminal.

`cd $HOME/leftwm/themes/Blue Coffee`
`chmod +x up`
`git commit up` (you'll add a commit comment to the file that comes up)
`git push` (You'll enter your github username and password)

**Comment by mautamu** [comment]
The issue that leftwm-theme needs to fix is to tell Git to download as blue_coffee / Blue_Coffee, and/or to go and upstream a fix for spaces in all themes with a space.

Former solution blocked on leftwm/leftwm-theme#9.

**Comment by VentGrey** [comment]
Blue Coffee is 404'd

**Comment by VuiMuich** [comment]
I guess we should revert back from submodules to directly committed themes. 
Maybe in addition simplify how to add additional repo sources to `leftwm-theme`?
